front/app/containers/Admin/projects/project/general/index.tsx

```diff
@@ -0,0 +1,662 @@
+import validateTitle from './utils/validateTitle';
+
+// typings
+import { IOption, Multiloc, UploadFile } from 'typings';
+import { isNilOrError } from 'utils/helperUtils';
+import useProjectFiles from 'hooks/useProjectFiles';
+import useProjectImages from 'hooks/useProjectImages';
+import { isEmpty, get, isString } from 'lodash-es';
+import eventEmitter from 'utils/eventEmitter';
+
+// services
+import {
+  IUpdatedProjectProperties,
+  addProject,
+  updateProject,
+} from 'services/projects';
+import { addProjectFile, deleteProjectFile } from 'services/projectFiles';
+import { addProjectImage, deleteProjectImage } from 'services/projectImages';
+
+// typings
+import { INewProjectCreatedEvent } from '../../all/CreateProject';
+
+export const TIMEOUT = 350;
+
+const AdminProjectsProjectGeneral = ({
+  intl: { formatMessage },
+}: InjectedIntlProps) => {
+  const params = useParams();
+  const localize = useLocalize();
+  const project = useProject({ projectId: params.projectId });
+  const appConfigLocales = useAppConfigurationLocales();
+  const remoteProjectFiles = useProjectFiles(params.projectId);
+  const remoteProjectImages = useProjectImages({
+    projectId: params.projectId || null,
+  });
+  const areas = useAreas();
+  const [submitState, setSubmitState] = useState<ISubmitState>('disabled');
+  const [processing, setProcessing] =
+    useState<IProjectFormState['processing']>(false);
+  const [apiErrors, setApiErrors] = useState({});
+  const [projectAttributesDiff, setProjectAttributesDiff] = useState<
+    IProjectFormState['projectAttributesDiff']
+  >({
+    admin_publication_attributes: {
+      publication_status: 'draft',
+    },
+  });
+  const [titleError, setTitleError] =
+    useState<IProjectFormState['titleError']>(null);
+  const [projectType, setProjectType] =
+    useState<IProjectFormState['projectType']>('timeline');
+  const [projectHeaderImage, setProjectHeaderImage] =
+    useState<IProjectFormState['projectHeaderImage']>(null);
+  const [projectFiles, setProjectFiles] = useState<
+    IProjectFormState['projectFiles']
+  >([]);
+  const [projectFilesToRemove, setProjectFilesToRemove] = useState<
+    IProjectFormState['projectFilesToRemove']
+  >([]);
+  const [projectImages, setProjectImages] = useState<
+    IProjectFormState['projectImages']
+  >([]);
+  const [projectImagesToRemove, setProjectImagesToRemove] = useState<
+    IProjectFormState['projectImagesToRemove']
+  >([]);
+  const [slug, setSlug] = useState<IProjectFormState['slug']>(null);
+  const [showSlugErrorMessage, setShowSlugErrorMessage] =
+    useState<IProjectFormState['showSlugErrorMessage']>(false);
+  const [publicationStatus, setPublicationStatus] =
+    useState<IProjectFormState['publicationStatus']>('draft');
+  const [areaType, setAreaType] =
+    useState<IProjectFormState['areaType']>('all');
+  const [areasOptions, setAreasOptions] = useState<
+    IProjectFormState['areasOptions']
+  >([]);
```

Comment:
This much state is probably a red flag. Better might be to keep it in child component or useReducer. But not going to refactor anymore now as it's out of scope.